### PR TITLE
Some fixes for the Awin implementation

### PR DESCRIPTION
**Pull request by rupertbates**
We have finally got the go ahead to continue with the affiliate tracking work that was started with https://github.com/guardian/subscriptions-frontend/pull/1093

This PR adds a tracking pixel and sets the `Awin` javascript object on `window` so that it can be found by their tracking script (it was getting lost before).

Related docs are [here](https://drive.google.com/drive/folders/19AOGPkPbbFjqPJTdCcCba3EwpLiSHBwW)

### CODE REVIEW

File: assets/javascripts/modules/analytics/awin.js [hunk 1]
```diff
@@ -11,14 +11,15 @@ define([
         AWIN.Tracking.Sale = {};
 
         /*** Set your transaction parameters ***/
-        const productData = guardian.pageInfo.productData
+        var productData = guardian.pageInfo.productData
```
Comment (davidfurey) [inline assets/javascripts/modules/analytics/awin.js]: Why did you need to change this to a `var`?

File: app/views/fragments/checkout/awinPixel.scala.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+@import views.support.PlanOps._
+@import com.gu.memsub.promo.Promotion.AnyPromotion
+
+@(subscription: Subscription[ContentSubscription], promotion: Option[AnyPromotion] = None)
+<!-- Image Pixel Tracking - Mandatory -->
+<img src="https://www.awin1.com/sread.img?tt=ns&tv=2
+&merchant=11152
+&amount=@subscription.firstPrice
+&cr=@subscription.currency
+&ref=@subscription.id.get
```
Comment (davidfurey) [inline app/views/fragments/checkout/awinPixel.scala.html]: Is `id` an option, should this have a getOrElse?

File: assets/javascripts/modules/analytics/awin.js (hunk 1 above)
Comment (svillafe) [inline assets/javascripts/modules/analytics/awin.js]: Why we can't keep `const` here?

Comment (rupertbates) [reply assets/javascripts/modules/analytics/awin.js]: I wasn't sure which js version we were targeting and Intellij was complaining. Digging a bit deeper though, it's fine so I've reinstated it.

Comment (rupertbates) [reply assets/javascripts/modules/analytics/awin.js]: I wasn't sure which js version we were targeting and Intellij was complaining. Digging a bit deeper though, it's fine so I've reinstated it.

Comment (svillafe) [reply assets/javascripts/modules/analytics/awin.js]: Ok great 👍

File: app/views/fragments/checkout/awinPixel.scala.html (hunk 2 above)
Comment (rupertbates) [reply app/views/fragments/checkout/awinPixel.scala.html]: It is not an option actually it just has a val called get

Comment (davidfurey) [reply app/views/fragments/checkout/awinPixel.scala.html]: Fair enough, slightly unexpected!
